File: lib-util/src/main/java/com/ustadmobile/core/impl/UmCallback.java
package com.ustadmobile.core.impl;

/**
 * Generic callback wrapper
 */

public interface UmCallback<T> {

    /**
     * Called when the operation completes successfully
     *
     * @param result result of the operation
     */
    void onSuccess(T result);

    /**
     * Called when the operation has failed
     *
     * @param exception exception thrown (if any)
     */
    void onFailure(Throwable exception);

}

File: lib-util/src/main/java/com/ustadmobile/core/impl/UmCallback.kt
package com.ustadmobile.core.impl

/**
 * Generic callback wrapper
 */

interface UmCallback<T> {

    /**
     * Called when the operation completes successfully
     *
     * @param result result of the operation
     */
    fun onSuccess(result: T?)

    /**
     * Called when the operation has failed
     *
     * @param exception exception thrown (if any)
     */
    fun onFailure(exception: Throwable)
}
